### PR TITLE
feat: average estimated processing fees

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
This PR is to support GroveDB v0.6 and estimated case costs.

## What was done?
Full support for estimated case costs.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive/Cargo.toml [hunk 1]
```diff
@@ -28,19 +28,20 @@ intmap = { version="2.0.0", features=["serde"] }
...
 dpp = { path = "../rs-dpp" }
-dashcore = { git="https://github.com/dashpay/rust-dashcore", features=["no-std", "secp-recovery", "rand", "signer"], default-features = false, branch="master" }
+itertools = { version = "0.10.5" }
+dashcore = { git="https://github.com/dashevo/rust-dashcore", features=["no-std", "secp-recovery", "rand", "signer"], default-features = false, branch="master" }
 
 [dependencies.grovedb]
-git = "https://github.com/dashpay/grovedb"
-rev = "df3535f1c26bec0539c6500d9d621eb4b18c61aa"
+git = "https://github.com/dashevo/grovedb"
+branch = "master"
```
Comment (shumkov) [inline packages/rs-drive/Cargo.toml:36]: Shall we use git tag?

File: packages/rs-drive/src/drive/document/estimation_costs.rs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+                KeyInfoPath::from_known_path(primary_key_path),
+                PotentiallyAtMaxElements(AllItems(
+                    DEFAULT_HASH_SIZE_U8,
+                    document_type.estimated_size() as u32,
+                    Some(flags_size),
+                )),
+            );
+        }
+    }
+    //
```
Comment (shumkov) [inline packages/rs-drive/src/drive/document/estimation_costs.rs:105]: Commented code

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive/Cargo.toml (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-drive/Cargo.toml:36]: Okay

**Review by shumkov — APPROVED** [review]
Don't really understand what is really written here but from the first perspective, it looks fine.

**Review by shumkov — APPROVED** [review]

**Pull request merged**